eq-author-api/scripts/createMigration.js

```diff
@@ -1,51 +1,65 @@
+/* eslint-disable no-console */
 const fs = require("fs").promises;
 const chalk = require("chalk");
 const { camelCase } = require("lodash");
 
-if (process.argv.length < 3) {
-  /* eslint-disable-next-line no-console */
-  console.error(
-    chalk.red(
-      "Not enough arguments provided to createMigration. Maybe the migration name is missing? Please refer to README."
-    )
-  );
-  process.exit(1);
-}
-
-const name = camelCase(process.argv[2]);
-const filename = `${name}.js`;
-const template = `//This is an auto-generated file.  Do NOT modify the method signature.
-
-module.exports = function ${name}(questionnaire) {
-  /**
-    [Insert migration here]
-   **/
-  return questionnaire;
-};
-`;
-
-const testTemplate = `const { cloneDeep } = require("lodash");
-const ${name} = require("./${filename}");
-
-describe("${name}", () => {
-  // This test must remain for your migration to always work
-  it("should be deterministic", () => {
-    const questionnaire = {}; // Fill in the structure of the questionnaire here
-    expect(${name}(cloneDeep(questionnaire))).toEqual(${name}(cloneDeep(questionnaire)));
-  });
+const MIGRATIONS_INDEX_PATH = "migrations/index.js";
 
-  it.todo("should...");
-});
+(async () => {
+  if (process.argv.length < 3) {
+    console.error(
+      chalk.red(
+        "Not enough arguments provided to createMigration. Maybe the migration name is missing? Please refer to README."
+      )
+    );
+    process.exit(1);
+  }
 
-`;
+  const name = camelCase(process.argv[2]);
+  const filename = `${name}.js`;
+  const template = `//This is an auto-generated file.  Do NOT modify the method signature.
+  
+  module.exports = function ${name}(questionnaire) {
+    /**
+      [Insert migration here]
+     **/
+    return questionnaire;
+  };
+  `;
 
-fs.writeFile(`migrations/${filename}`, template);
+  const testTemplate = `const { cloneDeep } = require("lodash");
+  const ${name} = require("./${filename}");
+  
+  describe("${name}", () => {
+    // This test must remain for your migration to always work
+    it("should be deterministic", () => {
+      const questionnaire = {}; // Fill in the structure of the questionnaire here
+      expect(${name}(cloneDeep(questionnaire))).toEqual(${name}(cloneDeep(questionnaire)));
+    });
+  
+    it.todo("should...");
+  });
+  
+  `;
 
-fs.writeFile(`migrations/${name}.test.js`, testTemplate);
+  await fs.writeFile(`migrations/${filename}`, template);
+  await fs.writeFile(`migrations/${name}.test.js`, testTemplate);
 
-/* eslint-disable-next-line no-console */
-console.info(
-  chalk.green(
-    `${name} migration successfully created. Remember to add this to 'migrations/index.js'`
-  )
-);
+  const currentMigrationList = await fs.readFile(MIGRATIONS_INDEX_PATH, "utf8");
+  const updatedMigrationList = currentMigrationList.replace(
+    /];/,
+    `  require('./${name}'),
```

Comment:
If create-migration is run with an existing migration name then a duplicate require statement is created.

Reply:
Yeah, we should probably reject that.